plugins/inputs/ssl_cert/ssl_cert.go

```diff
@@ -0,0 +1,166 @@
+package ssl_cert
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const sampleConfig = `
+  ## List of local SSL files
+  # files = []
+  ## List of servers
+  # servers = []
+  ## Timeout for SSL connection
+  # timeout = 5
+`
+const description = "Reads metrics from a SSL certificate"
+
+// SSLCert holds the configuration of the plugin.
+type SSLCert struct {
+	Servers []string      `toml:"servers"`
+	Files   []string      `toml:"files"`
+	Timeout time.Duration `toml:"timeout"`
+}
+
+// For tests
+var (
+	closeConn  bool
+	unsetCerts bool
+)
+
+// Description returns description of the plugin.
+func (sc *SSLCert) Description() string {
+	return description
+}
+
+// SampleConfig returns configuration sample for the plugin.
+func (sc *SSLCert) SampleConfig() string {
+	return sampleConfig
+}
+
+func getRemoteCert(server string, timeout time.Duration) (*x509.Certificate, error) {
+	tlsCfg := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+
+	ipConn, err := net.DialTimeout("tcp", server, timeout)
+	if err != nil {
+		return nil, err
+	}
+	defer ipConn.Close()
+
+	conn := tls.Client(ipConn, tlsCfg)
+	defer conn.Close()
+
+	if closeConn {
+		conn.Close()
+	}
+
+	hsErr := conn.Handshake()
+	if hsErr != nil {
+		return nil, hsErr
+	}
+
+	certs := conn.ConnectionState().PeerCertificates
+
+	if unsetCerts {
+		certs = nil
+	}
+
+	if certs == nil || len(certs) < 1 {
+		return nil, errors.New("couldn't get remote certificate")
+	}
+
+	return certs[0], nil
+}
+
+func getLocalCert(filename string) (*x509.Certificate, error) {
+	content, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	block, _ := pem.Decode(content)
+	if block == nil {
+		return nil, errors.New("failed to parse certificate PEM")
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return cert, nil
+}
+
+func getMetrics(cert *x509.Certificate, now time.Time) map[string]interface{} {
+	age := int(now.Sub(cert.NotBefore).Seconds())
+	expiry := int(cert.NotAfter.Sub(now).Seconds())
+	startdate := int(cert.NotBefore.Unix())
+	enddate := int(cert.NotAfter.Unix())
+
+	metrics := map[string]interface{}{
+		"age":       age,
+		"expiry":    expiry,
+		"startdate": startdate,
+		"enddate":   enddate,
+	}
```

Comment:
Ideally we could also have metrics for if the certificate is valid, including values for common errors such as the hostname not matching the certificate.  This is implemented in the related pr #3829, though ideally we could leverage the standard library code for this.